CovertInfiltration/Src/CovertInfiltration/Classes/UICovertActionsGeoscape.uc

```diff
@@ -802,25 +816,43 @@ simulated function UpdateCovertActionInfo()
 
 	DurationLabel.SetText(CurrentAction.bStarted ? class'UICovertActions'.default.CovertActions_TimeRemaining : class'UICovertActions'.default.CovertActions_Duration);
 	DurationValue.SetText(class'UIUtilities_Text'.static.AlignRight(CurrentAction.GetDurationString()));
-
+	
 	ExfiltrateLabel.SetText(strExfilLabel);
 	ExfiltrateValue.SetText(class'UIUtilities_Text'.static.AlignRight(class'UIUtilities_Strategy'.static.GetStrategyCostString(class'X2Helper_Infiltration'.static.GetExfiltrationCost(GetAction()), CostScalars)));
 	
+	ExpirationLabel.SetText(strExpiryLabel);
+	ExpirationValue.SetText(class'UIUtilities_Text'.static.AlignRight(string(GetExpirationDays())));
+	
 	if (GetAction().bStarted)
 	{
+		ExpirationLabel.Hide();
+		ExpirationValue.Hide();
+
 		ExfiltrateLabel.Show();
 		ExfiltrateValue.Show();
 	}
 	else
 	{
 		ExfiltrateLabel.Hide();
 		ExfiltrateValue.Hide();
+
+		ExpirationLabel.Show();
+		ExpirationValue.Show();
```

Comment:
This should be done only if the action is expiring. Also set the text only if it's expiring